Samples/Notifications/Push/cpp-console-unpackaged/cpp-console-unpackaged.cpp

```diff
@@ -0,0 +1,203 @@
+    auto kind = args.Kind();
+    switch (kind)
+    {
+        // When it is launched normally (by the users, or from the debugger), the sample requests a WNS ChannelURI and
+        // displays it, then waits for notifications. This user can take a copy of the WNS ChannelURI and use it to send
+        // notifications to the sample
+        case ExtendedActivationKind::Launch:
+        {
+            // Request a WNS ChannelURI which can be passed off to an external app to send notifications to.
+            // The WNS ChannelURI uniquely identifies this app for this user and device.
+            PushNotificationChannel channel = RequestChannel();
+
+            // Setup an event handler, so we can receive notifications in the foreground while the app is running.
+            if (channel)
+            {
+                SubscribeForegroundEventHandler(channel);
+
+                PushNotificationManager::Default().Register();
+            }
+            else
+            {
+                std::cout << "\nThere was an error obtaining the WNS ChannelURI" << std::endl;
+
+                if (remoteId == winrt::guid { "00000000-0000-0000-0000-000000000000" })
+                {
+                    std::cout << "\nThe remoteId has not been set. Refer to the readme file accompanying this sample\nfor the instructions to obtain and setup a remote id" << std::endl;
+                }
+            }
+
+            std::cout << "\nPress 'Enter' at any time to exit App." << std::endl;
+            std::cin.ignore();
+        }
+        break;
+
+        // When it is activated from a push notification, the sample only displays the notification.
+        // It doesn’t register for foreground activation of perform any other actions
+        // because background activation is meant to let app perform only small tasks in order to preserve battery life.
+        case ExtendedActivationKind::Push:
+        {
+            PushNotificationReceivedEventArgs pushArgs = args.Data().as<PushNotificationReceivedEventArgs>();
+
+            // Call GetDeferral to ensure that code runs in low power
+            auto deferral = pushArgs.GetDeferral();
+
+            auto payload = pushArgs.Payload();
+
+            // Do stuff to process the raw notification payload
+            std::string payloadString(payload.begin(), payload.end());
+            std::cout << "\nPush notification content received from BACKGROUND: " << payloadString.c_str() << std::endl;
+            std::cout << "\nPress 'Enter' to exit the App." << std::endl;
+
+            // Call Complete on the deferral when finished processing the payload.
+            // This removes the override that kept the app running even when the system was in a low power mode.
+            deferral.Complete();
+            std::cin.ignore();
+        }
+        break;
+
+        default:
+            // Unexpected activation type
+            std::cout << "\nUnexpected activation type" << std::endl;
```

Comment:
While we don't really care about all the other activation kinds, I think it would be valuable to log the ExtendedActivationKind instead of saying 'unexpected', but we just don't do anything with the activatedEventArgs.

Reply:
Actually, there's no ToString for the ExtendedActivationKind.